app/lib/features/events/pages/event_list_page.dart

```diff
@@ -0,0 +1,224 @@
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: _buildAppBar(),
+      body: _buildBody(),
+    );
+  }
+
+  AppBar _buildAppBar() {
+    return AppBar(
+      centerTitle: false,
+      title: Column(
+        mainAxisAlignment: MainAxisAlignment.start,
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: [
+          Text(L10n.of(context).events),
+          if (widget.spaceId != null) _buildSpaceName(),
+        ],
+      ),
+      actions: [
+        AddButtonWithCanPermission(
+          canString: 'CanPostEvent',
+          onPressed: () => context.pushNamed(
+            Routes.createEvent.name,
+            queryParameters: {'spaceId': widget.spaceId},
+          ),
+        ),
+      ],
+    );
+  }
+
+  Widget _buildSpaceName() {
+    String spaceName =
+        ref.watch(roomDisplayNameProvider(widget.spaceId!)).valueOrNull ?? '';
+    return Text(
+      '($spaceName)',
+      overflow: TextOverflow.ellipsis,
+      style: Theme.of(context).textTheme.labelLarge,
+    );
+  }
+
+  Widget _buildBody() {
+    AsyncValue<List<CalendarEvent>> eventList;
+    eventList = ref.watch(
+      eventListSearchFilterProvider(
+        (spaceId: widget.spaceId, searchText: searchValue),
+      ),
+    );
+
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.stretch,
+      children: [
+        _buildSearchBar(),
+        filterChipsButtons(),
+        Expanded(
+          child: eventList.when(
+            data: (events) => _buildEventList(events),
+            error: (error, stack) =>
+                Center(child: Text(L10n.of(context).loadingFailed(error))),
+            loading: () => const EventListSkeleton(),
+          ),
+        ),
+      ],
+    );
+  }
+
+  Widget _buildSearchBar() {
+    return Container(
+      margin: const EdgeInsets.symmetric(horizontal: 10, vertical: 15),
+      child: SearchBar(
+        controller: searchTextController,
+        leading: const Padding(
+          padding: EdgeInsets.all(8.0),
+          child: Icon(Atlas.magnifying_glass),
+        ),
+        hintText: L10n.of(context).search,
+        trailing: searchValue.isNotEmpty
+            ? [
+                IconButton(
+                  onPressed: () {
+                    FocusManager.instance.primaryFocus?.unfocus();
+                    ref.read(searchValueProvider.notifier).state = '';
+                    searchTextController.clear();
+                  },
+                  icon: const Icon(Icons.clear),
+                ),
+              ]
+            : null,
+        onChanged: (value) =>
+            ref.read(searchValueProvider.notifier).state = value,
+      ),
+    );
+  }
+
+  Widget filterChipsButtons() {
+    return SingleChildScrollView(
+      scrollDirection: Axis.horizontal,
+      child: Container(
+        padding: const EdgeInsets.symmetric(horizontal: 10, vertical: 4),
+        child: Row(
```

Comment:
don't we want to wrap if we don't have enough horizontal space (just thinking of languages with long texts instead)...
```suggestion
        child: Wrap(
```